### PR TITLE
Clk 1503 fix site imports reinstallations

### PR DESCRIPTION
**Pull request by stefan-hc**
I want this merged to master so I can re-point Clockwork codebase, fix Clockwork tests, and then - optionally - go back here to improve the code readability. After ensuring Clockwork does work with the HEAD of atlassian-jwt-authentication library.

### CODE REVIEW

File: lib/atlassian-jwt-authentication/verify.rb [hunk 1]
```diff
@@ -77,15 +79,15 @@ def verify
...
       rescue JWT::ExpiredSignature
         log(:error, "Error decoding JWT segments - signature is expired at #{data['exp']}")
-        return false
+        return UNVERIFIED_RESULT
       end
 
       unless header && payload
         log(:error, "Error decoding JWT segments - no header and payload for client_key #{data['iss']} and addon_key #{addon_key}")
-        return false
+        return UNVERIFIED_RESULT
```
Comment (skv-headless) [inline lib/atlassian-jwt-authentication/verify.rb:90]: It is golang style 😆 
Actually don't see a problem but try to raise an exception

**Review by skv-headless — APPROVED** [review]

File: lib/atlassian-jwt-authentication/verify.rb (hunk 1 above)
Comment (stefan-hc) [reply lib/atlassian-jwt-authentication/verify.rb:90]: I can try to clean this code up, but remember why we are not raising here (and elsewhere around) exceptions:
* we have two middlewares (one for Free, one for Pro), both verifying the same request
* I reckon that was the reason we were hesitant adding exceptions originating from .verify. Otherwise clockwork free was stopping to work...

I might be wrong though.

Comment (pawelniewie) [reply lib/atlassian-jwt-authentication/verify.rb:90]: > I might be wrong though.

So maybe check it?

File: lib/atlassian-jwt-authentication/filters.rb [hunk 2]
```diff
@@ -134,9 +134,9 @@ def _verify_jwt(addon_key, consider_param: false, skip_qsh_verification: false,
       jwt_verification.exclude_qsh_params = exclude_qsh_params
       jwt_verification.logger = logger if defined?(logger)
 
-      jwt_auth, account_id, context, qsh_verified = jwt_verification.verify
+      jwt_auth, account_id, context, qsh_verified, verified = jwt_verification.verify
```
Comment (pawelniewie) [inline lib/atlassian-jwt-authentication/filters.rb:137]: Too many *verified variables 🙈 

Can we differentiate them somehow with more meaningful names?

Comment (stefan-hc) [reply lib/atlassian-jwt-authentication/filters.rb:137]: Hard to say for me. Maybe another developer should make his/her hands dirty here and propose improvements.

The `jwt_verification.verify` - that guy is returning something representing verification status/result. I.e.: 
```
verification_result = jwt_verification.verify

if verification_result.failed(skip_qsh_validation)
  render_unauthorized
  return false
end

self.current_jwt_token = verification_result.jwt_auth
self.current_account_id = verification_result.account_id
self.current_jwt_context = verification_result.context
```
...but that would require creating a new class for that `verification_result`. And adjusting tests. And performing manual tests if Clockwork can still install under various scenarios. I don't know if I/we want to go that route...

File: lib/atlassian-jwt-authentication/verify.rb (hunk 1 above)
Comment (stefan-hc) [reply lib/atlassian-jwt-authentication/verify.rb:90]: Maybe not. My time is not unlimited (unfortunately), so I'd prefer to spend it on meaningful things instead. Tweaking the atlassian-jwt-authentication library for the sake of tweaking it is not something giving any value to our customers. Or to us to be honest.

If it works, don't touch it. Too critical component to play with.

I'll take a look here what can be improved, but I don't want to drag with this task more than 1 day. I/we've already spent more time than I've anticipated/wanted: [link](https://herocoders.atlassian.net/plugins/servlet/ac/clockwork-cloud/clockwork-timesheet?project.key=CLK&project.id=11304#!reportName=Clockwork%20Timesheet&scope%5BstartingAt%5D=2022-06-08&scope%5BendingAt%5D=2022-06-22&selectedBreakdowns%5B%5D=users&selectedBreakdowns%5B%5D=issues&selectedFilters%5BissueKeys%5D%5B%5D=CLK-1503&period=PERIOD_MONTH)
image

File: lib/atlassian-jwt-authentication/filters.rb (hunk 2 above)
Comment (pawelniewie) [reply lib/atlassian-jwt-authentication/filters.rb:137]: I'm not thinking about introducing some result object, but just changing the name. Looking at the code I don't get what the second verified is for.

File: lib/atlassian-jwt-authentication/verify.rb (hunk 1 above)
Comment (pawelniewie) [reply lib/atlassian-jwt-authentication/verify.rb:90]: I don't insist.

File: lib/atlassian-jwt-authentication/filters.rb (hunk 2 above)
Comment (stefan-hc) [reply lib/atlassian-jwt-authentication/filters.rb:137]: It's the overall verification result (true/false). So...

```
jwt_auth, account_id, context, qsh_verified, verification_result = jwt_verification.verify
```
Would that make more sense to you?

Comment (pawelniewie) [reply lib/atlassian-jwt-authentication/filters.rb:137]: Doesn't make a lot of difference. I'm leaving that to you. My brain hurts when looking at it. But I get that you want to get rid of this hot potato.

**Review by pawelniewie — APPROVED** [review]
Removing myself.

File: lib/atlassian-jwt-authentication/filters.rb (hunk 2 above)
Comment (stefan-hc) [reply lib/atlassian-jwt-authentication/filters.rb:137]: Yeah, exactly. My brains also hurts whenever I work with a-j-a library. It sure would benefit from refactoring or two. 

One of the reasons we don't do that is that a-j-a is not covered by tests enough. The only test in this repo is:
```
describe "RSpec" do
  it "actually works" do
    expect(1).to eq(1)
  end
end
```
:clap: :clap: :clap: :clapper:

And we have some coverage added with our tests in CLK repo. But they don't cover every possible combination and edge cases (think of double VerifyJwtToken middlewares used in the same app, think of various flows of installation/uninstallation/installation combined with free/pro versions). They don't cover even majority of flows I think.
 
So, to make refactoring a-j-a convenient and safe, we would have to put tremendous effort first to prepare a-j-a for such at-hoc refactorings. I mean, add a full blown suite of tests that would cover our asses in 99% cases. I don't want to lead that effort - it seems to me that more valuable to our customers will be if we just fix/plumb whatever we need in a-j-a and get back to our normal work ASAP.

**Comment by stefan-hc** [comment]
I'm going to proceed with merge, keeping in back of my head the suggestions/improvement requests you brought here. I just want first to stabilize heads of CLK+AJA, fix tests, and then go back to any further refactorings here.
